integrations/gcp/gcp_core/gcp_client.py

```diff
@@ -0,0 +1,109 @@
+from collections.abc import MutableSequence
+from typing import Any, Optional, TypeVar
+import proto  # type: ignore
+from google.api_core.exceptions import PermissionDenied, NotFound
+from google.cloud.asset_v1 import (
+    AssetServiceAsyncClient,
+    ContentType,
+)
+from google.cloud.asset_v1.services.asset_service import pagers
+from loguru import logger
+from google.pubsub_v1.services.publisher import PublisherAsyncClient
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE, RAW_ITEM
+
+RESOURCE_METADATA_CONTENT_TYPE = ContentType(1)
+GCP_PUBSUB_TOPIC_ASSET_KIND = "pubsub.googleapis.com/Topic"
+TOPIC_PROJECT_FIELD = "__project_id"
```

Comment:
Too specific key, lets set the full `__project` as extension to the resource so if user changes the way he defines identifiers he will be able to map the relations in the topic resource as well.